src/app/lab/components/dataset-toggle.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { Dataset } from '../../models/dataset.model';
+import { ProjectData } from '../../models/project-data.model';
+import { DatasetService } from '../../services/dataset.service';
+
+import * as _ from 'lodash';
+
+/*  Dataset Toggle Component
+
+    -- Requires project input
+    Expected use:
+        <ccl-dataset-toggle
+            [projectData]="your_project.project_data">
+*/
+
+@Component({
+  selector: 'ccl-dataset-toggle',
+  template: `
+    <div class="dropdown button-group">
+        <button class="button"
+               *ngFor="let s of datasets"
+               [disabled]="!checkValidDataset(s)"
+               [ngClass]="{'active': s.name === projectData.dataset.name}"
+               (click)="onDatasetSelected(s, $event)">
+            {{ s.name }}
+        </button>
+    </div>
+  `
+
+})
+export class DatasetToggleComponent implements OnInit {
+
+    @Input() projectData: ProjectData;
+    public datasets: Dataset[] = [];
+    private DEFAULT_DATASET_NAME = 'NEX-GDDP';
+
+    constructor(private datasetService: DatasetService) {}
+
+    ngOnInit() {
+        this.getDatasets();
+    }
+
+    public onDatasetSelected(dataset: Dataset, event?: Event) {
+        if (!this.isValidDataset(dataset)) {
+            this.selectDefaultDataset();
+        }
+        this.projectData.dataset = dataset;
+        this.projectData.models.forEach(model => {
+            model.enabled = _.includes(dataset.models, model.name);
+        });
+
+        if (event) {
+            event.preventDefault();
+        }
+    }
+
+    // set to default, or first valid option for the selected city
+    private selectDefaultDataset() {
+        let dataset = this.datasets.find(s => {
+            return s.name === this.DEFAULT_DATASET_NAME;
+        });
+
+        // if the standard default dataset is not valid, use first valid option
+        if (!this.isValidDataset(dataset)) {
+            dataset = this.datasets.find(s => {
+                return this.isValidDataset(s);
+            });
+        }
+
+        this.onDatasetSelected(dataset);
+    }
+
+    // check if a given dataset is valid and change it if not and it is selected
+    public checkValidDataset(dataset: Dataset): boolean {
+        const isValid = this.isValidDataset(dataset);
+        if (!isValid && this.projectData.dataset &&
+            this.projectData.dataset.name === dataset.name) {
+            // If this dataset is selected but not valid, change to a valid dataset.
```

Comment:
Do you have a workflow example such that I can test this behavior?

Reply:
In the 'Notes' section on this PR, there are instructions for adding (faking) a 'LOCA' dataset for a city. You can use that city in the lab to test out switching between datasets.

Specifically for this case:
 - select the city in the lab for which LOCA is valid, and select LOCA
 - change to a city for which LOCA is not valid, using the city picker in the upper left
 - the dataset selector should change back to NEX-GDP and LOCA should be greyed out

Reply:
This works. I ran into an issue bigger than this of checking that the scenario/dataset/city combo yields data or not. That check seems out of scope.

Reply:
I believe that a no data situation should only have occurred in testing this when a faked dataset was selected (see notes section). So this shouldn't be an issue.